pkg/tuned/cmd/render/render.go

```diff
@@ -262,3 +285,36 @@ func removeDuplicates[T string | int](sliceList []T) []T {
 	}
 	return list
 }
+
+func findMachineConfigPoolByName(mcPools []*mcfgv1.MachineConfigPool, mcpName string) *mcfgv1.MachineConfigPool {
+	for _, mcp := range mcPools {
+		if mcp.Name == mcpName {
+			return mcp
+		}
+	}
+	return nil
+}
+
+func filterPerformanceProfilesByMachineConfigPool(performanceProfiles []*performancev2.PerformanceProfile, mcp *mcfgv1.MachineConfigPool) ([]*performancev2.PerformanceProfile, error) {
+	mcpSelector, err := metav1.LabelSelectorAsSelector(mcp.Spec.NodeSelector)
```

Comment:
I think we should instead call the perf profile code that computes the machine config labels and then compare it with the MCP.machineConfigSelector.

https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile/components/profile/profile.go#L26

Reply:
I have used NodeSelector because we are using NodeSelector in the NTO code to relate MCPs and PerformanceProfiles
[here](https://github.com/openshift/cluster-node-tuning-operator/blob/8069ccec36de1699373941a25fa08c6f282ebe26/pkg/performanceprofile/controller/performanceprofile_controller.go#L190) to reconcile the PerformanceProfiles affected by a change in a MCP
and [here](https://github.com/openshift/cluster-node-tuning-operator/blob/8069ccec36de1699373941a25fa08c6f282ebe26/pkg/performanceprofile/controller/performanceprofile_controller.go#L792) to look for the MCPs when we need to reconcile a PerformanceProfile

Reply:
Yep, per our discussion lets make the method consistent across the project and figure out whether it is the right approach outside of this PR.